source/adios2/ADIOS.h

```diff
@@ -97,13 +97,34 @@ class ADIOS
      * @param dimensions
      * @param selections
      * @param offsets
+     * @constantShape true if dimensions, offsets and local sizes don't change
+     * over time
      * @return reference to Variable object
      */
     template <class T>
-    Variable<T> &DefineVariable(const std::string &name,
-                                const Dims localDimensions = Dims{1},
-                                const Dims globalDimensions = Dims{},
-                                const Dims offsets = Dims{});
+    Variable<T> &DefineVariable(const std::string &name, const Dims shape = {},
+                                const Dims start = {}, const Dims count = {},
+                                bool constantShape = false);
+
+    template <class T>
+    Variable<T> &DefineLocalValue(const std::string &name)
```

Comment:
Not a fan, the less public API functions to remember the better...just like MPI or OpenMP core functionality. How is the returned Variable<T>& different from DefineVariable<T>(name) and DefineLocalValue<T>(name)?   I understand the separation for DefineVariableCompound as it requires more member functions to set the offsets, but not for DefineVariable<T>.   As a user I don't like remembering more than a few things, otherwise they think it's "complicated".